src/Hazelcast.Net/Clustering/Authenticator.cs

```diff
@@ -268,4 +272,43 @@ internal MemberGroups ParsePartitionMemberGroups(ClientAuthenticationCodec.Respo
 
         return new MemberGroups(partitionGroups, version, response.ClusterId, response.MemberUuid);
     }
+
+    internal IDictionary<CPGroupId, Guid> ParseCPGroupLeaderIds(ClientAuthenticationCodec.ResponseParameters response)
+    {
+        var cpGroupLeaderIds = new Dictionary<CPGroupId, Guid>();
+        
+        if (!response.IsKeyValuePairsExists)
+            return cpGroupLeaderIds;
+        
+        var isContainsCPGroupLeaderIds = response.KeyValuePairs.TryGetValue(ClusterCPGroups.CPGroupsJsonField, out var jsonMessage);
+        
+        if (isContainsCPGroupLeaderIds && string.IsNullOrEmpty(jsonMessage))
+        {
+            return cpGroupLeaderIds;
+        }
+        
+        try
+        {
+            var groupsObject = JsonNode.Parse(jsonMessage);
+            
+            foreach (var group in groupsObject.AsArray())
+            {
+                var cpGroupJson = group["raftId"];
+                var seed = cpGroupJson["seed"].GetValue<int>();
+                var id = cpGroupJson["id"].GetValue<int>();
+                var name = cpGroupJson["name"].GetValue<string>();
+                var leaderId = group["leaderUUID"].GetValue<Guid>();
+
+                var cpGroupId = new CPGroupId(name, seed, id);
+                cpGroupLeaderIds[cpGroupId] = leaderId;
+            }
+        }
+        catch (Exception e)
+        {
+            _logger.IfDebug()?.LogDebug("Failed to parse CP group leader ids [{JsonMessage}]: {Message}. ", jsonMessage, e.Message);
```

Comment:
Should this not be > `DEBUG`? We control the structure of the message and its content -- if it doesn't have what we expect then it's an `ERROR`?